pkgs/by-name/xe/xeve/package.nix

```
@@ -17,18 +17,33 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-8jXntm/yFme9ZPImdW54jAr11hEsU1K+N5/7RLmITPs=";
   };
 
+  patches = [
+    ./0001-xeve_enc-Drop-if-without-body-nor-assignment.patch
+    ./0002-xevem_util-Avoid-skipping-cnt_tmp-values.patch
+    ./0003-xeve_app_util-Darwin-is-UNIX-like-enough.patch
+    ./0004-sse2neon-Cast-to-variable-type.patch
+    ./0005-pkgconfig-Use-CMAKE_INSTALL_FULL_.patch
+    ./0006-CMakeLists.txt-Conditionally-append-C-flags.patch
+    ./0007-CMakeLists.txt-Disable-static-linking-on-Darwin.patch
+    ./0008-xevem_alf.h-Drop-typedef-redefinition-for-C99-compat.patch
+  ];
+
   postPatch = ''
     echo v$version > version.txt
   '';
 
   nativeBuildInputs = [ cmake ];
 
+  cmakeFlags = let inherit (lib) cmakeBool cmakeFeature optional;
+                   inherit (stdenv.hostPlatform) isAarch64 isDarwin;
+                in optional isAarch64 (cmakeBool "ARM" true)
+                ++ optional isDarwin
+                            (cmakeFeature "CMAKE_SYSTEM_NAME" "Darwin");
```

```suggestion
  cmakeFlags = lib.optional stdenv.hostPlatform.isAarch64 (lib.cmakeBool "ARM" true)
    ++ lib.optional stdenv.hostPlatform.isDarwin (lib.cmakeFeature "CMAKE_SYSTEM_NAME" "Darwin");
```
less and boring code is always better :)

---

It's an existing pattern in Nixpkgs, the systemd stuff in nixos/modules makes heavy use of it.
It's just as explicit as fully qualified names and it makes the code more readable IMO so I'd prefer not to drop this.